Error loading tests with regex in package.json

**Describe the bug**
- [ ] test load missing suites/tests
- [x] test load errors
- [ ] test running errors

A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
(I am not sure if you will be able to reproduce the error, but hopefully this will help)
1. Have a package.json in the root of the project
2. Put jest configuration in a `package.json`:
```json
{
    ... // other project settings
    "jest": {
    "modulePaths": [
      "<rootDir>/build/js/"
    ],
    "setupTestFrameworkScriptFile": "<rootDir>/tests/js/testFrameworkScriptFile.js",
    "testRegex": ".*\\.js"  // In my case regex is a little bit more complicated, but I am facing an error even with this one 
  }
}
```
3. Put tests in the `tests/js/`
4. Try to reload tests
5. Get an error:
```
[2020-09-17 13:56:51.935] [ERROR] Error while reloading all tests. [TypeError: e.testRegex.some is not a function
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2535079
	at Array.some (<anonymous>)
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2534814
	at t.default.<anonymous> (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2533881)
	at Generator.next (<anonymous>)
	at o (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:16:1418)]
```

**Expected behavior**
Tests are explored correctly and I can see them in the tests tab

**Desktop (please complete the following information):**
 - OS: OSX

**Additional context**
Here is a full output from the extension when I try to reload tests:
```
[2020-09-17 13:56:51.803] [INFO] Loading Jest tests...
[2020-09-17 13:56:51.805] [INFO] Loading Jest settings from /Users/pavel/projects/eb/package.json...
[2020-09-17 13:56:51.932] [INFO] More than one Jest test regex found. {
  jestVersionMajor: 23,
  configs: [
    {
      automock: false,
      browser: false,
      cache: true,
      cacheDirectory: '/var/folders/0_/6w94lk1171vc_fpwbjqx02nw0000gn/T/jest_dx',
      clearMocks: false,
      coveragePathIgnorePatterns: [Array],
      detectLeaks: false,
      detectOpenHandles: false,
      errorOnDeprecated: false,
      filter: null,
      forceCoverageMatch: [],
      globals: {},
      haste: [Object],
      moduleDirectories: [Array],
      moduleFileExtensions: [Array],
      moduleNameMapper: {},
      modulePathIgnorePatterns: [],
      modulePaths: [Array],
      name: '523afd3e38b2345a8f12a6a37c91da53',
      prettierPath: null,
      resetMocks: false,
      resetModules: false,
      resolver: null,
      restoreMocks: false,
      rootDir: '/Users/pavel/projects/eb',
      roots: [Array],
      runner: 'jest-runner',
      setupFiles: [Array],
      setupTestFrameworkScriptFile: '/Users/pavel/projects/eb/tests/js/testFrameworkScriptFile.js',
      skipFilter: false,
      snapshotSerializers: [],
      testEnvironment: 'jest-environment-jsdom',
      testEnvironmentOptions: {},
      testLocationInResults: false,
      testMatch: [],
      testPathIgnorePatterns: [Array],
      testRegex: '.*\\.js',
      testRunner: '/Users/pavel/projects/eb/node_modules/jest-cli/node_modules/jest-jasmine2/build/index.js',
      testURL: 'http://localhost',
      timers: 'real',
      transform: [Array],
      transformIgnorePatterns: [Array],
      watchPathIgnorePatterns: []
    }
  ]
}
[2020-09-17 13:56:51.933] [INFO] Force loading all tests...
[2020-09-17 13:56:51.933] [INFO] Loading Jest tests
[2020-09-17 13:56:51.935] [ERROR] Error while reloading all tests. [TypeError: e.testRegex.some is not a function
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2535079
	at Array.some (<anonymous>)
	at /Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2534814
	at t.default.<anonymous> (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:425:2533881)
	at Generator.next (<anonymous>)
	at o (/Users/pavel/.vscode/extensions/kavod-io.vscode-jest-test-adapter-0.7.2/dist/extension.js:16:1418)]
[2020-09-17 13:56:51.935] [INFO] Results:
[2020-09-17 13:56:51.935] [INFO] Finished loading Jest tests.
```

The extension says that `More than one Jest test regex found`, but I don't see any other regexes. 
I've tried to modify my regex from `package.json` and neither seem to work for me.
However, if I remove regex, the extension loads tests successfully (but more than I need, because I am using regex to exclude some portion of tests).

I am happy to provide any additional information if necessary.

Thanks!

If I change my `testRegex` to an array, then I am getting following error:
```
[2020-09-18 05:41:44.870] [ERROR] Error loading tests '● Validation Error:\n' +
  '\n' +
  '  Option "testRegex" must be of type:\n' +
  '    string\n' +
  '  but instead received:\n' +
  '    array\n' +
  '\n' +
  '  Example:\n' +
  '  {\n' +
  '    "testRegex": "(/__tests__/.*|(\\\\.|/)(test|spec))\\\\.jsx?$"\n' +
  '  }\n' +
  '\n' +
  '  Configuration Documentation:\n' +
  '  https://jestjs.io/docs/configuration.html\n' +
  '\n'
```

My jest version is `jest@25.2.3`

--- a/src/JestSettings/__tests__/index.test.ts
+++ b/src/JestSettings/__tests__/index.test.ts
@@ -1,0 +1,69 @@
+import { JestSettings } from "jest-editor-support";
+import _ from "lodash";
+import { getSettings } from "../index";
+
+let mockSettings: JestSettings;
+
+jest.mock("jest-editor-support", () => ({
+  getSettings: () => mockSettings,
+}));
+
+describe("getSettings tests", () => {
+  it(`Given a Jest config with a string for testRegex
+      when getSettings is called
+      then testRegex returns a string array with a single value`, async () => {
+    mockSettings = createConfigWithTestRegex("some regex");
+    // @ts-ignore
+    const workspace: ProjectWorkspace = {};
+
+    const settings = await getSettings(workspace);
+
+    expect(settings.configs[0].testRegex).toStrictEqual(["some regex"]);
+    compareConfigsExcludingTestRegex(settings, mockSettings);
+  });
+
+  it(`Given a Jest config with a string array for testRegex
+      when getSettings is called
+      then testRegex returns a string array with a single value`, async () => {
+    mockSettings = createConfigWithTestRegex(["some regex", "another regex"]);
+    // @ts-ignore
+    const workspace: ProjectWorkspace = {};
+
+    const settings = await getSettings(workspace);
+
+    expect(settings.configs[0].testRegex).toStrictEqual(["some regex", "another regex"]);
+    expect(settings).toStrictEqual(mockSettings);
+  });
+});
+
+const compareConfigsExcludingTestRegex = (actual: JestSettings, expected: JestSettings) => {
+  const actualEntries = Object.entries(actual.configs[0]).filter(([key]) => key !== "testRegex");
+  const expectedEntries = Object.entries(expected.configs[0]).filter(([key]) => key !== "testRegex");
+
+  expect(actualEntries).toHaveLength(expectedEntries.length);
+
+  for (let i = 0; i < actualEntries.length; i++) {
+    const a = actualEntries[i];
+    const b = expectedEntries[i];
+
+    expect(a[0]).toBe(b[0]);
+    expect(a[1]).toStrictEqual(b[1]);
+  }
+};
+
+const createConfigWithTestRegex = (testRegex: string[] | string): JestSettings => {
+  return {
+    configs: [
+      // @ts-ignore
+      {
+        automock: true,
+        browser: false,
+        cwd: "somewhere",
+        name: "test project",
+        // @ts-ignore
+        testRegex,
+      },
+    ],
+    jestVersionMajor: 20,
+  };
+};

--- a/src/JestSettings/index.ts
+++ b/src/JestSettings/index.ts
@@ -7,8 +7,26 @@ import {
 } from "jest-editor-support";
 import { ProjectConfig } from "../repo";
 
-const getSettings = (projectConfig: ProjectConfig): Promise<JestSettings> => {
-  return editorSupportGetSettings(convertToWorkspace(projectConfig));
+const getSettings = async (projectConfig: ProjectConfig): Promise<JestSettings> => {
+  const rawEditorSettings = await editorSupportGetSettings(convertToWorkspace(projectConfig));
+
+  // modern Jest should return us `testRegex` values that are string[], but we just fix up anything that comes through
+  // as a single string.
+  const configs = rawEditorSettings.configs.map(c => {
+    if (isString(c.testRegex)) {
+      return ({...c, testRegex: [c.testRegex as string]});
+    }
+    return c;
+  });
+
+  return {
+    ...rawEditorSettings,
+    configs,
+  };
+};
+
+const isString = (value: any): value is string => {
+  return typeof value === "string" || value instanceof String;
 };
 
 const createRunner = (projectConfig: ProjectConfig, options: Options) => {

